### ISSUE TITLE
"No hits" shown incorrectly on print statement

### ISSUE THREAD

**Issue by bhackett1024** [RecordReplay/devtools#6549]
This seems to happen when adding a print statement before indexing finishes.  Replay with comments:

https://app.replay.io/recording/invalid-no-hits--8c51609c-f563-4f53-8857-dba1005ee64a?point=546164725844869027945326080604256823&time=232105&hasFrames=false

**Comment by jasonLaster** [comment]
@jcmorrow tagged you with the current sprint because I assume this is related to your current focus on hitCounts, printstatements, focus, loading

**Comment by jcmorrow** [comment]
This recording keeps dying so it's really hard to tell what is going on 😞

**Comment by bhackett1024** [comment]
It looks like the recording is dying because of https://github.com/recordreplay/backend/issues/5076, the server process is crashing due to what looks like an inconsistency in the scan data.  I'll try to figure out what's going on there.

**Comment by jcmorrow** [comment]
This is also from this recording: https://www.loom.com/share/ccd949b58f3c4d269a8b704def77b919

**Comment by jcmorrow** [comment]
I made https://github.com/RecordReplay/backend/issues/5414 for it.

**Comment by jasonLaster** [comment]
@jcmorrow i can imagine there is a timing issue when print statements are queued up before indexing is finished

**Comment by jcmorrow** [comment]
Yeah, so this is a good example of https://www.notion.so/replayio/Loading-Region-Problems-b8dc0d03c49840f8b504e967c7ea5778. We need to be smarter about when we requery things. It's not enough to say "Wait for the entire recording to load before you issue an analysis". Because sometimes analyses will overflow and be rejected - and we need to re-run those upon changing the focus window (because after that they might work again).

**Comment by jcmorrow** [comment]
I left a comment on http://localhost:8080/recording/invalid-no-hits--8c51609c-f563-4f53-8857-dba1005ee64a?point=39591263552052737907188651427627580&time=13953.572424181111&hasFrames=true where I finally found the issue. It was *very* hard to track down because of lots of *other* issues I was having with breakpoints.

**Comment by jcmorrow** [comment]
I'm fairly sure that this particular issue is solved now, I've stress-tested by creating a bunch of controllers and asking for analyses immediately, and they properly wait to be loaded. I think having everyone on turbo replays means we will not see this anymore, but if we do, let's make sure to catalogue and reopen!
